lib/Backend/arm64/LowerMD.cpp

```diff
@@ -1506,40 +1495,35 @@ LowererMD::LowerEntryInstr(IR::EntryInstr * entryInstr)
         if (r17Opnd == nullptr || isScratchRegisterThrashed)
         {
             r17Opnd = r17Opnd ? r17Opnd : IR::RegOpnd::New(nullptr, SCRATCH_REG, TyMachReg, this->m_func);
-            // mov r12, 0
-            IR::Instr * instrMov = IR::Instr::New(Js::OpCode::MOV, r17Opnd, IR::IntConstOpnd::New(0, TyMachReg, this->m_func), this->m_func);
+            // mov r17, 0
+            IR::Instr * instrMov = IR::Instr::New(Js::OpCode::MOVZ, r17Opnd, IR::IntConstOpnd::New(0, TyMachReg, this->m_func), this->m_func);
```

Comment:
Again, watch the limits of MOVZ (0-65535). You should at least have an Assert in place here to catch it in case.

Reply:
Only moving 0, should be ok?

---
In reply to: [143539114](https://github.com/Microsoft/ChakraCore/pull/3903#discussion_r143539114) [](ancestors = 143539114)

Reply:
Yes, that's fine. Sorry still getting used to reading the syntax of these and missed that it was a constant 0.